crates/accelerate/src/synthesis/evolution/mod.rs

```diff
@@ -0,0 +1,439 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2024
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+#![allow(clippy::too_many_arguments)]
+
+use crate::synthesis::clifford::greedy_synthesis::resynthesize_clifford_circuit;
+
+use pyo3::prelude::*;
+use pyo3::types::{PyList, PyString, PyTuple};
+use smallvec::{smallvec, SmallVec};
+use std::borrow::Borrow;
+
+use qiskit_circuit::circuit_data::CircuitData;
+use qiskit_circuit::operations::{multiply_param, radd_param, Param, StandardGate};
+use qiskit_circuit::Qubit;
+
+use rustiq_core::structures::{
+    CliffordCircuit, CliffordGate, IsometryTableau, Metric, PauliLike, PauliSet,
+};
+use rustiq_core::synthesis::clifford::isometry::isometry_synthesis;
+use rustiq_core::synthesis::pauli_network::greedy_pauli_network;
+
+use rustworkx_core::petgraph::graph::NodeIndex;
+use rustworkx_core::petgraph::prelude::StableDiGraph;
+use rustworkx_core::petgraph::Incoming;
+
+/// A Qiskit gate. The quantum circuit data returned by the pauli network
+/// synthesis algorithm will consist of clifford and rotation gates.
+type QiskitGate = (StandardGate, SmallVec<[Param; 3]>, SmallVec<[Qubit; 2]>);
+
+/// Expands the sparse pauli string representation to the full representation.
+///
+/// For example: for the input `sparse_pauli = "XY", qubits = [1, 3], num_qubits = 6`,
+/// the function returns `"IXIYII"`.
+fn expand_pauli(sparse_pauli: String, qubits: Vec<u32>, num_qubits: usize) -> String {
```

Comment:
Ah, in my code I was passing the ownership to `qubits` when calling `expand_pauli`, which compiled since `qubits` were not used after that. But agree that passing this as a slice is a far better API. Done (the commit will soon follow).